app/src/androidTest/java/com/github/se/eventradar/signup/SignupTest.kt

```diff
@@ -1,163 +1,157 @@
-    fun profilePictureIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        profilePicture { assertIsDisplayed() }
+      eventRadarLogo {
+        performScrollTo()
+        assertIsDisplayed()
       }
-    }
-
-    /** This test checks that the username text field is correctly displayed */
-    @Test
-    fun usernameTextFieldIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        usernameTextField { assertIsDisplayed() }
+      profilePicture {
+        performScrollTo()
+        assertIsDisplayed()
       }
-    }
-
-    /** This test checks that the name text field is correctly displayed */
-    @Test
-    fun nameTextFieldIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        nameTextField { assertIsDisplayed() }
+      usernameTextField {
+        performScrollTo()
+        assertIsDisplayed()
       }
-    }
-
-    /** This test checks that the surname text field is correctly displayed */
-    @Test
-    fun surnameTextFieldIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        surnameTextField { assertIsDisplayed() }
+      nameTextField {
+        performScrollTo()
+        assertIsDisplayed()
       }
-    }
-
-    /** This test checks that the phone text field is correctly displayed */
-    @Test
-    fun phoneTextFieldIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        phoneTextField { assertIsDisplayed() }
+      surnameTextField {
+        performScrollTo()
+        assertIsDisplayed()
       }
-    }
-
-    /** This test checks that the birth date text field is correctly displayed */
-    @Test
-    fun birthDateTextFieldIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        birthDateTextField { assertIsDisplayed() }
+      phoneTextField {
+        performScrollTo()
+        assertIsDisplayed()
+      }
+      birthDateTextField {
+        performScrollTo()
+        assertIsDisplayed()
+      }
+      signUpButton {
+        performScrollTo()
+        assertIsDisplayed()
+        assertHasClickAction()
       }
     }
+  }
 
-    /** This test checks that the sign up button is correctly displayed */
-    @Test
-    fun signUpButtonIsCorrectlyDisplayed() {
-      ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
-        // Test the UI elements
-        signUpButton {
+  /** This test checks that the google authentication works correctly */
+  @Test
+  fun googleSignInReturnsValidActivityResult() = run {
+    Intents.init()
+    ComposeScreen.onComposeScreen<SignupScreen>(composeTestRule) {
+      step("fill in all fields") {
```

Comment:
nit: Consider adding helper functions to reduce code redundancy. You can use them for both SignupTest.kt and SignupFailureTest.kt.